File: calc-sheet-column-number/calc-sheet-column-number.java
class Main {
   
   public static int convertTitleToNumber(String title) {
      int converted = 0;
      int pow = 1;
      
      for(int i = title.length() - 1; i >= 0; i--) {
         converted += pow * ((int)title.charAt(i) - 64);
         pow *= 26;
      }
      
      return converted;
   }
   
   public static void main(String[] args) {
      String[] titles = {"A", "AB", "ZY", "AAA", "B", "AC"};

      for(String s : titles)
         System.out.println(s + " => " + convertTitleToNumber(s));
   }

}

File: calc-sheet-column-number/calc-sheet-column-number.kt

fun convertTitleToNumber(title: String): Int{
    val base: Int = 26
    var converted: Int = 0
    var temp: Int = 0

    for (i in title.length downTo 1){
        converted += (title[i-1].toInt()-64) * calculatePow(temp,base)
        temp += 1
    }
    return converted
}

fun calculatePow(exponent: Int, base: Int): Int {
    return (Math.pow(base.toDouble(), exponent.toDouble())).toInt()
}


fun main(args: Array<String>) {
    println("Enter text: ")
    val stringInput: String = readLine()!!
    println("$stringInput :  ${convertTitleToNumber(stringInput)}")
}

